lib/neovim-plugin.nix

```diff
@@ -104,16 +119,20 @@ with lib;
       config =
         let
           cfg = config.${namespace}.${name};
-          extraConfigNamespace = if isColorscheme then "extraConfigLuaPre" else "extraConfigLua";
+          extraConfigNamespace = if isColorscheme then "configLuaPre" else "configLua";
         in
         mkIf cfg.enable (mkMerge [
           {
             extraPlugins = (optional installPackage cfg.package) ++ extraPlugins;
             inherit extraPackages;
 
-            ${extraConfigNamespace} = optionalString callSetup ''
+            ${namespace}.${name}.${extraConfigNamespace} = optionalString callSetup ''
               require('${luaName}')${setup}(${optionalString (cfg ? settings) (toLuaObject cfg.settings)})
             '';
+
+            extraConfigLua = cfg.configLua;
+            extraConfigLuaPre = cfg.configLuaPre;
+            extraConfigLuaPost = cfg.configLuaPost;
```

Comment:
From reading the option names, I was actually expecting to see the config grouped together in `extraConfigLua`:

```nix
extraConfigLua = concatStringsSep "\n" [
  cfg.configLuaPre
  cfg.configLua
  cfg.configLuaPost
];
```

Allowing users to add config before/with/after the plugin's internal config.

I think this is more useful for end-users, but I can see use cases for both approaches.

Maybe we should have `preConfigLua{Pre,Post,}`, `configLua{Pre,Post,}`, and `postConfigLua{Pre,Post,}`?

Though this feels a bit cumbersome...

Reply:
Yeah this feels a bit complicated. I hesitated on the two approaches, but I prefered going globaly, as it implies the local approach. We can always revisit this behaviour later, for example by adding a global option `localPluginConfig` that choses what to do.
This a bit in the spirit of what is made availaible by this PR!

Reply:
Ignoring any internal usage we have for a second, I think from a user perspective the fact the options are nested under `plugins.<name>` implies they are related to that plugin.\
Therefore I think the "local" pre/post makes most sense to users.

I'm hesitant to merge as-is if we're already considering how it may change going forward, given we may end up running into rename conflicts (etc).

---

I wonder if we can mitigate the confusion (and the clunky-ness of having 9 options) by renaming the "global" options.

Maybe the top-level options could be something like `extraConfig{Early,Late,}` (dropping the implied `Lua` and changing from pre/post to early/late)?

More local options would then be free to use "pre" and "post" without confusion, while falling back to "early" and "late" when they need to tie in with the more "global" config locations?